subsys/bluetooth/services/fast_pair/fp_advertising.c

```diff
@@ -113,8 +176,13 @@ static int fp_adv_data_fill_non_discoverable(struct net_buf_simple *buf, size_t
 		__ASSERT_NO_MSG(ak_filter_size <= BIT_MASK(LEN_BITS));
 		net_buf_simple_add_u8(buf, ENCODE_FIELD_LEN_TYPE(ak_filter_size, ak_filter_type));
 
-		err = fp_crypto_account_key_filter(net_buf_simple_add(buf, ak_filter_size), ak,
-						   account_key_cnt, salt, NULL);
+		if (fp_adv_info.adv_battery_mode != BT_FAST_PAIR_ADV_BATTERY_MODE_NONE) {
+			err = fp_crypto_account_key_filter(net_buf_simple_add(buf, ak_filter_size),
+							   ak, account_key_cnt, salt, battery_info);
```

Comment:
I saw it's already fixed in a proper way :)

Reply:
If you wish, you can introduce additional boolean to avoid copy-pasting this long comparison:
`bool add_battery_info = (fp_adv_config.adv_battery_mode != BT_FAST_PAIR_ADV_BATTERY_MODE_NONE)`